File: flutter_video_plugin/android/src/main/java/com/jd/fluttervideoplugin/VideoViewController.java
package com.jd.fluttervideoplugin;

import android.content.Context;
import android.view.View;

import java.util.Map;

import androidx.annotation.NonNull;
import io.flutter.Log;
import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.platform.PlatformView;


public class VideoViewController implements PlatformView, MethodChannel.MethodCallHandler {

    private static final String NATIVE_CCTV_VIEW_TYPE_ID = "flutter_video_controller";
    private final BaseVideoViewController videoView;
    private Context context;
    private Map<String, Object> params;

    VideoViewController(Context context, BinaryMessenger messenger, int id, Map<String, Object> params) {
        this.context = context;
        this.params = params;
        BaseVideoViewController view = BaseVideoViewController.alloc();
        this.videoView = view;
        //注册消息通道
        MethodChannel methodChannel = new MethodChannel(messenger, NATIVE_CCTV_VIEW_TYPE_ID+"_" + id);
        methodChannel.setMethodCallHandler(this);
    }

    @Override
    public View getView() {
        return this.videoView.getView(this.context,this.params);
    }

    @Override
    public void dispose() {

    }

    @Override
    public void onMethodCall(@NonNull MethodCall methodCall, @NonNull MethodChannel.Result result) {
        String method = methodCall.method;
        if ("prepare".equals(method)) {
            this.videoView.prepare();
            result.success(null);
        } else if ("play".equals(method)) {
            this.videoView.play();
            result.success(null);
        } else if ("pause".equals(method)) {
            this.videoView.pause();
            result.success(null);
        } else if ("stop".equals(method)) {
            this.videoView.stop();
            result.success(null);
        } else if ("resume".equals(method)) {
            this.videoView.resume();
            result.success(null);
        } else if ("replay".equals(method)) {
            this.videoView.replay();
            result.success(null);
        } else if ("mute".equals(method)) {
            this.videoView.mute();
            result.success(null);
        } else if ("unmute".equals(method)) {
            this.videoView.unmute();
            result.success(null);
        } else if ("seekTo".equals(method)) {
            double text = (double) methodCall.arguments;
            this.videoView.seekTo(text);
            result.success(null);
        } else if ("duration".equals(method)) {
            String duration = this.videoView.duration();
            result.success(duration);
        } else if ("setRate".equals(method)) {
            String text = (String) methodCall.arguments;
            this.videoView.setRate(text);
            result.success(null);
        }
    }
}

File: flutter_video_plugin/ios/Classes/VideoViewController.m
#import "VideoViewController.h"
#import "BaseVideoViewController.h"

@interface VideoViewController ()

@property(nonatomic, strong) BaseVideoViewController *videoView;
@property(nonatomic) int64_t viewId;
@property(nonatomic, strong) FlutterMethodChannel *channel;

@property(nonatomic, strong) NSDictionary *params;

@end

@implementation VideoViewController

- (instancetype)initWithWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args binaryMessenger:(NSObject<FlutterBinaryMessenger> *)messenger{
    if ([super init]) {
        self.params = args;
        self.videoView = [BaseVideoViewController newInstance];
        //注册消息通道
        self.viewId = viewId;
        NSString *channelName = [NSString stringWithFormat:@"flutter_video_controller_%lld", viewId];
        self.channel = [FlutterMethodChannel methodChannelWithName:channelName binaryMessenger:messenger];
        __weak __typeof__(self) weakSelf = self;
        [self.channel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult  result) {
            [weakSelf onMethodCall:call result:result];
        }];
    }

    return self;
}

- (UIView *)view {
    return [self.videoView getView:self.params];
}

- (void)onMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result{
    
    NSString *method = call.method;
    
    if ([@"prepare" isEqualToString:method]) {
        [self.videoView prepare];
        result(nil);
    } else if ([@"play" isEqualToString:method]) {
        [self.videoView play];
        result(nil);
    } else if ([@"pause" isEqualToString:method]) {
        [self.videoView pause];
        result(nil);
    } else if ([@"stop" isEqualToString:method]) {
        [self.videoView stop];
        result(nil);
    } else if ([@"resume" isEqualToString:method]) {
        [self.videoView resume];
        result(nil);
    } else if ([@"replay" isEqualToString:method]) {
        [self.videoView replay];
        result(nil);
    } else if ([@"mute" isEqualToString:method]) {
        [self.videoView mute];
        result(nil);
    } else if ([@"unmute" isEqualToString:method]) {
        [self.videoView unmute];
        result(nil);
    } else if ([@"seekTo" isEqualToString:method]) {
        NSString *text = call.arguments;
        [self.videoView seekTo:text.doubleValue];
        result(nil);
    } else if ([@"duration" isEqualToString:method]) {
         NSString *duration = [self.videoView duration];
        result(duration);
    } else if ([@"setRate" isEqualToString:method]) {
        NSString *rate = call.arguments;
        [self.videoView setRate:rate];
        result(nil);
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end
